pkg/tests/schema_pulumi_test.go

```diff
@@ -4159,3 +4159,100 @@ func TestMakeTerraformResultNilVsEmptyMap(t *testing.T) {
 		assert.True(t, props["test"].DeepEquals(emptyMap))
 	})
 }
+
+func TestTypeChecker(t *testing.T) {
+	t.Setenv("PULUMI_DEBUG_YAML_DISABLE_TYPE_CHECKING", "true")
+	resMap := map[string]*schema.Resource{
+		"prov_test": {
+			Schema: map[string]*schema.Schema{
+				"tags": {
+					Type:     schema.TypeMap,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+					Optional: true,
+				},
+				"network_configuration": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"assign_public_ip": {
+								Type:     schema.TypeBool,
+								Optional: true,
+								Default:  false,
+							},
+							"security_groups": {
+								Type:     schema.TypeSet,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+							"subnets": {
+								Type:     schema.TypeSet,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	runTest := func(t *testing.T, resMap map[string]*schema.Resource, props interface{}, expectedError string) {
+		propsJSON, err := json.Marshal(props)
+		require.NoError(t, err)
+		program := fmt.Sprintf(`
+name: test
+runtime: yaml
+resources:
+  mainRes:
+    type: prov:index:Test
+	properties: %s`, propsJSON)
+
+		bridgedProvider := pulcheck.BridgedProvider(t, "prov", &schema.Provider{ResourcesMap: resMap})
+		pt := pulcheck.PulCheck(t, bridgedProvider, program)
+		_, err = pt.CurrentStack().Up(pt.Context())
+
+		require.Error(t, err)
```

Comment:
Perhaps I missed this but does the type-checker error by default now? Or are we still in the mode where it was generating warnings. Just to make sure it's the type-checker catching this.

Reply:
Yeah, fair - it is a warning but something else fails in the provider. I'll change this to assert on the warning instead.

Reply:
Out of scope for this PR but think about this if you are interested in this topic, I think we're very conservative with introducing new warnings but we don't assert on that in the tests. Possibly something in the test framework could be turned to strict mode by default to insist on no warnings or else on no new warnings except expected ones.

Reply:
I've seen TF tests have such an env var which turns all warnings into failures - we could add something like that for CI. It'd also give us additional testing on these type checker warnings as we'd see any failures in provider CI